src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1001,10 +1001,14 @@ function ($a) {
                         ->arrayNode('middlewares')
                             ->addDefaultsIfNotSet()
                             ->children()
-                            ->arrayNode('doctrine_transaction')
-                                ->canBeEnabled()
-                                ->children()
-                                    ->scalarNode('entity_manager_name')->info('The name of the entity manager to use')->defaultNull()->end()
+                                ->arrayNode('doctrine_transaction')
+                                    ->canBeEnabled()
+                                    ->children()
+                                        ->scalarNode('entity_manager_name')->info('The name of the entity manager to use')->defaultNull()->end()
+                                    ->end()
+                                ->end()
+                                ->arrayNode('validation')
+                                    ->canBeDisabled()
```

Comment:
```diff
- ->canBeDisabled()
+ ->{!class_exists(FullStack::class) && class_exists(Validation::class) ? 'canBeDisabled' : 'canBeEnabled'}()
```

so it does not appear as enabled when the validator component isn't installed?

Reply:
Excellent idea. I was looking for something like this. And if someone tries to use validation without the validation component I'll throw an exception.

Reply:
Indeed, we use to throw such exceptions from the FrameworkExtension :)